src/Core/Services/Implementations/EmergencyAccessService.cs

```diff
@@ -53,9 +54,9 @@ public class EmergencyAccessService : IEmergencyAccessService
             _mailService = mailService;
             _userService = userService;
             _passwordHasher = passwordHasher;
-            _dataProtector = dataProtectionProvider.CreateProtector("EmergencyAccessServiceDataProtector");
             _globalSettings = globalSettings;
             _organizationService = organizationService;
+            _tokenizer = tokenizerFactory.Create<EmergencyAccessInviteToken>("", TokenType.DataProtector);
```

Comment:
Wondering if we can replace this with injecting the dependency `ITokenizer<EmergencyAccessInviteToken>` directly. Similar to how asp.net injects `Ilogger<class>`. https://stackoverflow.com/questions/39029344/factory-pattern-with-open-generics

Reply:
The factory is to enable `clearTextPrefix`, not the generic. I don't think we can DI inject a class-specific constant to the constructor of an `ITokenizer`

Reply:
@MGibson1 Is there any scenario where we'll use a token type with different prefixes? If not I think we should move the `TokenType.DataProtector` and `clearTextPrefix` into the factory somehow. I dislike that the services should have this knowledge since at least to me this feels like implementation details of how the token works.

Reply:
Done